Add britepool userid submodule

bansawbanchee opened the pull request:
<!--
Thank you for your pull request. Please make sure this PR is scoped to one change, and that any added or changed code includes tests with greater than 80% code coverage. See https://github.com/prebid/Prebid.js/blob/master/CONTRIBUTING.md#testing-prebidjs for documentation on testing Prebid.js.
-->

## Type of change
<!-- Remove items that don't apply and/or select an item by changing [ ] to [x] -->
- [ ] Bugfix
- [ ] Feature
- [ ] New bidder adapter  <!--  IMPORTANT: if checking here, also submit your bidder params documentation here https://github.com/prebid/prebid.github.io/tree/master/dev-docs/bidders --> 
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [X] Other

## Description of change
Adding BritePool User Id Module

- contact email of the adapter’s maintainer
- [x] prebid@britepool.com

For any changes that affect user-facing APIs or example code documented on http://prebid.org, please provide:

- https://github.com/prebid/prebid.github.io/pull/1559

## Other information
Docs pull request
https://github.com/prebid/prebid.github.io/pull/1559

bretg commented:
@bansawbanchee - please fix conflicts. We recently merged two other user ID sub-modules, so I suspect that your branch wasn't up-to-date.

bansawbanchee commented:
@jdwieland8282 @jlukas79 we updated the pull request as asked. We also split our individual tests out into our own spec file and updated the docs site which is linked above. Let us know if we need to do anything else.

bansawbanchee commented:
@jdwieland8282 went ahead and added the support for prebid server as requested athttps://github.com/prebid/prebid.github.io/pull/1559#issuecomment-546067692 also updated those docs to sho support for prebid server.

bansawbanchee commented:
@jdwieland8282 @jlukas79 any update on getting this into a release?

pm-harshad-mane commented:
Hello @bansawbanchee , 
Can you add a commit to restart the CI job? just add a comment in code and commit to your branch.

bansawbanchee commented:
> Hello @bansawbanchee ,
> Can you add a commit to restart the CI job? just add a comment in code and commit to your branch.

@pm-harshad-mane job still failing.

pm-harshad-mane commented:
Hello @bansawbanchee and @idettman , @jaiminpanchal27 @jsnellbaker 
I ran CI job 2 times, now it is passed.

bansawbanchee commented:
@jdwieland8282 @jlukas79 following up on this now that builds are passing.

bansawbanchee commented:
@pm-harshad-mane  Anything we can do to get this merged? It's been sitting here for quite some time now.

Thanks.

jlukas79 commented on modules/userId/index.js:
```diff
@@ -106,6 +106,7 @@ import CONSTANTS from '../../src/constants.json';
 import {module} from '../../src/hook';
 import {unifiedIdSubmodule} from './unifiedIdSystem.js';
 import {pubCommonIdSubmodule} from './pubCommonIdSystem.js';
+import {britepoolIdSubmodule} from './britepoolIdSystem.js';
```
Please note in PreBid 3.0 unifiedId, and pubCommonId will be separate modules not installed by default with user ID module itself.  Please remove the default installation.

cfaherty replied:
Hi. I've moved our module out of the default installation.

bretg commented:
@jsnellbaker - up for the final review here?

cfaherty commented:
@bretg @jsnellbaker Hi. Is there anything we still need to do to help this along?

jsnellbaker commented on modules/britepoolIdSystem.md:
```diff
@@ -0,0 +1,42 @@
+                url: 'https://sandbox-api.britepool.com/v1/britepool/id', // optional
+                api_key: ’xxx’, // provided by britepool
+                hash: ’yyyy’, // example identifier
+                ssid: 'r894hvfnviurfincdejkencjcv' // example identifier
```
Can you update the (majority) of the single quotes used around the values here?  They look like back-tics (`) rather than single quotes (')?

cfaherty replied:
Fixed.

cfaherty replied:
Also I've added a sandbox working api_key and identifiers in the britepoolIdSystem.md
